src/main/scala/logger/Logger.scala

```diff
@@ -54,7 +54,7 @@ trait LazyLogging {
   * when used in multi-threaded environments
   */
 private class LoggerState {
-  var globalLevel = LogLevel.None
+  var globalLevel = LogLevel.Warn
```

Comment:
I'm fine with this as is, but you may consider using the default value in the annotation to try to get at one source for ground truth here.

```suggestion
  var globalLevel = LogLevelAnnotation.globalLogLevel
```

Reply:
Hmm. The doesn't appear to compile:
```
Error:(57, 40) value globalLogLevel is not a member of object logger.LogLevelAnnotation
  var globalLevel = LogLevelAnnotation.globalLogLevel
```
It seems I would need an instance of the class `LogLevelAnnotation`. Do you have one handy?

Reply:
No, we'd have to construct one (non ideal). Maybe we can add this to the object:

```diff
diff --git a/src/main/scala/logger/LoggerAnnotations.scala b/src/main/scala/logger/LoggerAnnotations.scala
index 323b46e5..f8d01d32 100644
--- a/src/main/scala/logger/LoggerAnnotations.scala
+++ b/src/main/scala/logger/LoggerAnnotations.scala
@@ -14,10 +14,12 @@ sealed trait LoggerOption { this: Annotation => }
   *  - if unset, a [[LogLevelAnnotation]] with the default log level will be emitted
   * @param level the level of logging
   */
-case class LogLevelAnnotation(globalLogLevel: LogLevel.Value = LogLevel.None) extends NoTargetAnnotation with LoggerOption
+case class LogLevelAnnotation(globalLogLevel: LogLevel.Value = LogLevelAnnotation.default) extends NoTargetAnnotation with LoggerOption
 
 object LogLevelAnnotation extends HasShellOptions {
 
+  val default = LogLevel.Warn
+
   val options = Seq(
     new ShellOption[String](
       longOption = "log-level",

```

Reply:
I'm slightly concerned about the coupling between logging and annotations. In my mind, logging is a short step up from `println`. I realize it may be convenient to have access to options/annotations for generating logging scopes, but my gut feeling is to invert the dependency and have logging at a lower level (with no awareness of annotations or options).

Reply:
Note: I view this narrowly about defining a single source of truth for the default logging level. I'd be fine if that lives in the logger state and that default is used in the log level annotation.

For stages, it's proved convenient to define defaults in the annotations because those also define the option parsing. And it seemed natural to put parsing and defaults together.

I still approve of the PR as is. :+1:

Reply:
Right. I agree that parsing and defaults are connected. I'm not sure I have a clear concept of how `LoggerState` should be managed.

Note that prior to this PR there were two independent defaults for `LogLevelAnnotation` and `LoggerState`. I'd like to leave their merging to a discussion of who's on top and a subsequent PR.